features/nanostack/FEATURE_NANOSTACK/nanostack_entropy_check.cpp

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2015 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbedtls/platform.h"
+
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

Comment:
As @sbutcher-arm pointed out, the problem is that `MBEDTLS_ENTROPY_HARDWARE_ALT` at this stage is like "internal" to mbed TLS. However, if you include "mbedtls/config.h" `MBEDTLS_ENTROPY_HARDWARE_ALT` and `DEVICE_TRNG` are interchangeable. If you would like to be consistent with the `target.json` then I would suggest you use `DEVICE_TRNG` though.